scRNA-seq-advanced/05-aucell.Rmd

```diff
@@ -0,0 +1,403 @@
+
+## Set up gene sets
+
+We are going to use two gene sets pertaining to Ewing sarcoma.
+
+* [`ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION`](https://www.gsea-msigdb.org/gsea/msigdb/geneset_page.jsp?geneSetName=ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION), which are genes that were highly expressed in a rhabdomyosarcoma cell line engineered to express the EWSR1-FLI1 fusion.
+* [`RIGGI_EWING_SARCOMA_PROGENITOR_UP`](https://www.gsea-msigdb.org/gsea/msigdb/cards/RIGGI_EWING_SARCOMA_PROGENITOR_UP), which are genes that were highly expressed in mesenchymal stem cells engineered to express the EWS-FLI1 fusion protein.
+
+We would expect both of these gene sets to have high expression in tumor cells.
+
+```{r genesets}
+ewing_gene_set_names <- c("ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION",
+                          "RIGGI_EWING_SARCOMA_PROGENITOR_UP")
+```
+
+These gene sets come from the C2 gene set collection from MSigDB.
+Let's retrieve them using `msigdbr()`.
+
+```{r extract_genesets, live = TRUE}
+ewing_gene_sets_df <- msigdbr(species = "Homo sapiens",
+                              category = "C2",
+                              subcategory = "CGP") |>
+  dplyr::filter(gs_name %in% ewing_gene_set_names)
+```
+
+`AUCell` uses gene sets in a particular format that comes from the `GSEABase` package.
+We need to create a `GeneSetCollection`.
+
+```{r gene_set_collection}
+ewing_gene_set_collection <- ewing_gene_set_names |>
+  purrr::map(
+    # For each gene set
+    \(gene_set_name) ewing_gene_sets_df |>
+      # Subset to the rows in that gene set
+      dplyr::filter(gs_name == gene_set_name) |>
+      # Grab the Ensembl gene identifiers
+      dplyr::pull(ensembl_gene) |>
+      # Create a GeneSet object
+      GeneSet(setName = gene_set_name,
+              geneIdType = ENSEMBLIdentifier())
+  ) |>
+  # Turn the list of GeneSet objects into a GeneSet collection
+  GeneSetCollection()
+```
+
+## Read in and prepare SingleCellExperiment
+
+```{r read_in_sce, live = TRUE}
+sce <- readr::read_rds(sce_file)
+```
+
+The `AUCell` functions takes an expression matrix with genes as rows and cells as column.
+We can save a counts matrix in sparse format for use with `AUCell`.
+
+```{r sparse_matrix}
+# Extract counts matrix and save in sparse format
+counts_matrix <- counts(sce) |>
+  as("dgCMatrix")
+```
+
+There may be genes in our gene set that do not appear in the SingleCellExperiment object.
+We can remove them using the `subsetGeneSets()` function.
+
+```{r subset_gene_sets, live = TRUE}
+# Remove genes from gene sets if they are not in the SCE
+ewing_gene_set_collection <- subsetGeneSets(ewing_gene_set_collection,
+                                            rownames(counts_matrix))
+```
+
+## AUCell
+
+AUCell relies on gene rankings, but we expect a proportion of genes are not detected.
+Genes can also have the same expression level (i.e., ties).
+These undetected genes and ties will be randomly ordered.
+To make our results reproducible, we will set a seed.
```

Comment:
Same comment about expanding on the text a little bit. I would have a little intro here that talks about the method itself and then at the end state that because ties will be randomly ordered we first need to set a seed.